src/main/java/nl/ordina/jobcrawler/controller/JobcrawlerController.java

```diff
@@ -0,0 +1,80 @@
+package nl.ordina.jobcrawler.controller;
+
+import nl.ordina.jobcrawler.SearchRequest;
+import nl.ordina.jobcrawler.SearchResult;
+import nl.ordina.jobcrawler.model.Vacancy;
+import nl.ordina.jobcrawler.controller.exception.VacancyNotFoundException;
+import nl.ordina.jobcrawler.service.VacancyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.*;
+
+
+@CrossOrigin(origins = "http://localhost:4200")
+@RestController
+public class JobcrawlerController {
+
+    @Autowired
+    private VacancyService vacancyService;
+
+    // Post mappings
+
+    @PostMapping("/searchrequest")
+    public SearchResult searchRequest(@RequestBody SearchRequest request) {
+
+        // Temporary mock data
+        String requestText = "For request: " + request.getLocation() + ", " + request.getDistance() + ", " + request.getKeywords() +  ":";
+        String[] texts = {requestText, "Test result 1", "Test result 2"};
+        String[] links = {"http://google.com/", "http://duckduckgo.com/"};
+
+        return new SearchResult(texts, links);
+
+    }
+
+    @PostMapping("/addJobWithJson")
+    public Vacancy addJob(@Valid @RequestBody Vacancy job) {
+        return vacancyService.add(job);
+    }
+
+    // Get mappings
+
+    @GetMapping("/getByID/{id}")
+    public Vacancy getByID(@PathVariable("id") UUID id) {
+        // Retrieve a vacancy by its ID (UUID). If Vacancy is not found it throws a 'VacancyNotFoundException' (HttpStatus.NOT_FOUND).
+        Optional<Vacancy> vacancy = vacancyService.getByID(id);
+
+        if(!vacancy.isPresent())
+            throw new VacancyNotFoundException("Vacancy with id: " + id + " not found.");
+
+        return vacancy.get();
```

Comment:
```suggestion
return vacancy.orElseThrow(() -> new VacancyNotFoundException("Vacancy with id: " + id + " not found."));
```

Reply:
Checking whether an optional is present is bad practice. 

Helaas was hier een beetje Lambda magic nodig, maar deze manier heeft wel de voorkeur.